implementations/rust/vault/src/types.rs

```diff
@@ -293,6 +222,8 @@ pub enum SecretKey {
     Curve25519([u8; 32]),
     /// NIST P-256 (secp256r1, prime256v1) secret key
     P256([u8; 32]),
+    /// ChainKey
+    ChainKey([u8; 32]),
```

Comment:
This feels like the opposite of modularity. ChainKey is a key exchange term. Key exchange terms shouldn't bleed into vault.

Reply:
To which category from existing ones does ChainKey goes then?